### DISCUSSION TITLE
esphom to HAA OTA

### DISCUSSION THREAD

**Discussion by YouCanNotBeSerious** [RavenSystem/esp-homekit-devices#2149] [General]
In case it helps someone else,  I was in the situation of having a hard to access esphome sonoff basic device and found that it can be converted to a HAA device easily!

Simply upload haamain.bin using the esphome device web server. I guess they use the same OTA mechanism.  Maybe the fact it is so easy to change should be put in the wiki please?

**Comment by RavenSystem** [comment]
Doing this, you break some HAA parts, OTA updates included.

**Reply by YouCanNotBeSerious** [reply]
Oh OK, thanks for the clarifcation!   My test device which is sonoff basic R2 seems ok from limited time testing...but since it's installed with the latest firmware I have not needed or tried OTA update yet.  _Is there anything else that should be a problem, if i never OTA update it?_

What about the method of esphome to tasmota, then HAA? I see that method documented in the [installation page](https://github.com/RavenSystem/esp-homekit-devices/wiki/Installation)....any disadvantage?   

I do understand the best idea is to flash it properly with serial port, I'm trying to save some access problems in a wall :-)
